### ISSUE TITLE
`weave expose -h` doesn't register with weaveDNS if there is an existing address

### ISSUE THREAD

**Issue by awh** [weaveworks/weave#1626]
On current `master`:
```
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave launch
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave expose -h foo.weave.local
10.32.0.1
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave status dns
vagrant@vagrant-ubuntu-vivid-64:~/weave$
```
However on `1.2` branch:
```
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave launch
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave expose -h foo.weave.local
10.32.0.1
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave status dns
foo          10.32.0.1       weave:expose 4a:25:30:35:57:ef
```
Very surprised there is no smoke test for this - I'm pretty sure there was one, perhaps it got lost in a test refactoring. We should add one with the fix.

**Comment by awh** [comment]
False alarm - I had already run `weave expose` once without `-h`. See
```
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave reset
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave launch
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave expose -h foo.weave.local
10.32.0.1
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave status dns
foo          10.32.0.1       weave:expose 1a:45:d3:32:58:2b
```
vs:
```
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave reset
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave launch
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave expose
10.32.0.1
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave expose -h foo.weave.local
10.32.0.1
vagrant@vagrant-ubuntu-vivid-64:~/weave$ ./weave status dns
vagrant@vagrant-ubuntu-vivid-64:~/weave$
```

**Comment by awh** [comment]
Confirmed in `1.2` as well.

**Comment by awh** [comment]
Closed by 94984df047dc2a32abc6d842519434c0fd7cd703 (why is GitHub not automatically closing these branch PRs when they get merged to master?)
